<a href="https://colab.research.google.com/github/bengosney/Advent-Of-Code-2018/blob/master/AdventOfCode_2018_day_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Advent of Code - Day 1

## The input
Nothing to see here...

In [0]:
with open('day1') as f:
  data = f.read()
  
data = data.splitlines()

## Part One

### Attempt One

In [2]:
freq = 0

for change in data:
  freq += int(change)
  
freq

520

Got the correct answer but it's quite long, I can do better

### Attempt Two

In [3]:
sum([int(change) for change in data])

520

Single line, much better

## Part Two

### Attempt one
I didn't read it properly... 

> *Note that your device might need to repeat its list of frequency changes many times before a duplicate frequency is found*



In [4]:
import collections

freq = 0
hist = []
for change in data:
  freq += int(change)
  hist.append(freq)
  
print([item for item, count in collections.Counter(hist).items() if count > 1])

[]


yea, didn't work, so I went back read the task. I also didn't like using an external lib
### Attempt two

In [5]:
freq = 0
hist = []
first = None
loop = 0
while first == None:
  loop += 1
  for change in data:
    freq += int(change)
  
    if freq in hist and first == None:
      first = freq
  
    hist.append(freq)
    
    
print("Fount {} on loop {}".format(first, loop))

Fount 394 on loop 140


This took *a lot* longer than I expected to run the 140 loops but got the correct answer. 
And it's a lot of code.